oauthenticator/generic.py

```diff
@@ -89,7 +89,9 @@ def authenticate(self, handler, data=None):
         params = dict(
             redirect_uri=self.get_callback_url(handler),
             code=code,
-            grant_type='authorization_code'
+            grant_type='authorization_code',
+            client_id=self.client_id,
```

Comment:
Can you maybe gate adding these on `if not self.basic_auth`? I think it makes sense, otherwise. Basic auth is used by default.